metrics/metrics.go

```diff
@@ -38,6 +39,30 @@ const (
 	ReshareShutdown      ReshareStatus = "node_stopped"
 )
 
+type statsHandler struct {
```

Comment:
this whole type is an `IncomingConnectionsStatsHandler` because it tags to the `IncomingConnectionTimestamp` metric.

Consider moving this statsHandler struct over to the `net` package, so that when e.g. the http client package pulls in metrics, it doesn't take on the grpc dependency.

Reply:
good call, thanks! Fixed.